sympy/solvers/solveset.py

```diff
@@ -1133,13 +1137,162 @@ def _is_exponential(f, symbol):
+
+    symbol : Symbol
+        The variable in which the equation is solved
+
+    Returns
+    =======
+
+    A simpler equation containing a single instance of log.
+
+    Examples
+    ========
+
+    >>> from sympy import symbols, S, log, Eq
+    >>> from sympy.solvers.solveset import \
+    ... _solve_logarithm as solve_log
+    >>> x = symbols('x')
+    >>> f = Eq(log(x - 3) + log(x + 3), 0)
+    >>> solve_log(f, x)
+    log((x - 3)*(x + 3))
+
+    * Proof of correctness
+
+    A logarithm is another way to write exponent and is defined by
+
+    .. math:: {log_b x} = y \enspace if \enspace b^y = x
+
+    When one side of the equation contains a single logarithm, the
+    equation can be solved by rewriting the equation as an equivalent
+    exponential equation as defined above. But if one side contains
+    more than one logarithm, we need to use the properties of logarithm
+    to condense it into a single logarithm.
+
+    Take for example
+
+    .. math:: log(2*x) - 15 = 0
+
+    contains single logarithm, therefore we can directly rewrite it to
+    exponential form as
+
+    .. math:: x = exp(15)/2
+
+    But if the equation has more than one logarithm as
+
+    .. math:: log(x - 3) + log(x + 3) = 0
+
+    we use log identities to convert into a reduced form
+
+    Using,
+
+    .. math:: log(a) + log(b) = log(a*b)
+
+    therefore the equation becomes,
+
+    .. math:: log((x - 3)*(x  + 3))
+
+    This equation contains one logarithm and can be solved by rewriting
+    to exponents.
+    """
+
+    lhs, rhs = f.lhs, f.rhs
+    new_lhs = logcombine(lhs, force=True)
+    new_f = new_lhs - rhs
+
+    return new_f
+
+
+def _is_logarithm(f, symbol):
+    r"""
+    Helper to check whether the equation is logarithmic or not.
+
+    Parameters
+    ==========
+
+    f : Expr
+        The equation to be checked as an instance of `Eq`
+
+    symbol : Symbol
+        The variable in which the equation is checked
+
+    Returns
+    =======
+
+    `True` if the equation is logarithmic otherwise `False`.
+
+    Examples
+    ========
+
+    >>> from sympy import symbols, tan, log, Eq
+    >>> from sympy.solvers.solveset import _is_logarithm as check
+    >>> x = symbols('x')
+    >>> check(log(x + 2) - log(x + 3), x)
+    True
+    >>> check(tan(log(2*x)), x)
+    False
+
+    * Philosophy behind the helper
+
+    The function extracts each term and checks whether it is
+    logarithmic w.r.t `symbol`.
+    """
+    rv = False
+    for term in Add.make_args(f):
+        saw_log = False
+        for term_arg in _term_factors(term):
+            if symbol not in term_arg.free_symbols:
+                continue
+            if isinstance(term_arg, log):
+                if not term_arg.args[0].has(log):
+                    if saw_log:
+                        return False  # more than one log in term
+                    saw_log = True
+                else:
+                    if _is_logarithm(term_arg.args[0], symbol):
```

Comment:
The trick and beauty of recursion is to let someone else make the decisions. This function should not need to introspect: solveset will figure out what to do with the transformed expression which was logarithmic...bt not logarithmic in a polynomial of x.